test_conformance/images/clFillImage/test_fill_generic.cpp

```diff
@@ -79,26 +161,63 @@ cl_mem create_image( cl_context context, cl_command_queue queue, BufferOwningPtr
             log_error( "ERROR: Unable to create backing store for pitched 3D image. %ld bytes\n",  imageInfo->depth * imageInfo->slicePitch );
             return NULL;
         }
-        mem_flags = CL_MEM_READ_ONLY | CL_MEM_USE_HOST_PTR;
+        if (imageInfo->type != CL_MEM_OBJECT_IMAGE1D_BUFFER)
+        {
+            mem_flags = CL_MEM_READ_ONLY | CL_MEM_USE_HOST_PTR;
+        }
     }
 
-    img = clCreateImage(context, mem_flags, imageInfo->format, &imageDesc, host_ptr, error);
+    if (imageInfo->type != CL_MEM_OBJECT_IMAGE1D_BUFFER)
+    {
+        img = clCreateImage(context, mem_flags, imageInfo->format, &imageDesc,
+                            host_ptr, error);
+    }
+    else
+    {
+        img = clCreateImage(context, mem_flags, imageInfo->format, &imageDesc,
+                            nullptr, error);
+    }
 
     if (gEnablePitch)
     {
-        if ( *error == CL_SUCCESS )
+        struct pitch_buffer_data *data = (struct pitch_buffer_data *)malloc(
+            sizeof(struct pitch_buffer_data));
+        data->buf = host_ptr;
+        data->is_aligned = CL_VERSION_MAJOR(version) == 1
```

Comment:
We believe this needs to be:

-        data->is_aligned = CL_VERSION_MAJOR(version) == 1
-            && imageInfo->type == CL_MEM_OBJECT_IMAGE1D_BUFFER;
+        data->is_aligned =  !((CL_VERSION_MAJOR(version) == 1)
+            || (imageInfo->type != CL_MEM_OBJECT_IMAGE1D_BUFFER));

Reply:
```
data->is_aligned = !((CL_VERSION_MAJOR(version) == 1)
|| (imageInfo->type != CL_MEM_OBJECT_IMAGE1D_BUFFER));
```
does not match 
> Actual: if version != 1 and type is 1Dbuffer then buffer is aligned

I think `version != 1 && type is 1DBuffer` is correct. I'll push that modification.